### DISCUSSION TITLE
django3 branch: files only play when web browser tab has focus

### DISCUSSION THREAD

**Discussion by kratcliff** [ScanOC/trunk-player#118] [General]
I decided to try the django3 branch, and now I've got an issue that's probably my fault but I'm not sure where I went wrong or how to resolve it:

When I listen to my site everything seems to work fine until I minimize my web browser or click on another program or even switch to another browser tab. Then the trunk-player tab no longer plays new transmissions. As soon as I bring the web browser tab back into focus it starts playing audio automatically from from where it left off.

This occurs in both Chrome and Firefox on Windows and linux. Neither browser gives any pertinent information in the developer console.

Any ideas? Thanks for your help, and for the continued development of trunk-player!

**Comment by bctrainers** [comment]
This behavior is sort of built into chromium and firefox, it's meant to save on CPU i/o - thus less power utilization. TrunkPlayer uses a rather dated version of jPlayer, and likely does not have the capability to persist loading in the background properly.

On chrome, you can check `chrome://discards/` and see if the TrunkPlayer tab is being actively discarded. You can toggle it there, but not sure if that is persistent.

**Comment by MaxwellDPS** [comment]
@kratcliff Thanks for letting me know on this, I will be reworking the ui to use ReactJS here, I will make sure to implement a modern player

**Reply by kratcliff** [reply]
Thanks for the replies and explanation, @bctrainers and @MaxwellDPS!

It's exciting to see new development happening with the project. I can't help write code, but I am willing to test new code on my system and report back.
